appsmithorg/appsmith
# chore: defer registration of instance to when required

## Description
- Currently, the code to register Appsmith instance is part of `switchIfEmpty` block which eagerly evaluates the fallback stream irrespective of whether the main stream is available or not. This causes the log `Triggering registration of this instance...` log to always appear on the console even if the registration has already happened. This PR updates the `switchIfEmpty` block to lazily evaluate the fallback stream by using `Mono.defer`. 
- [Ref1](https://stackoverflow.com/questions/57870706/whats-the-point-of-switchifempty-getting-evaluated-eagerly), [Ref2](https://stackoverflow.com/questions/54373920/mono-switchifempty-is-always-called#:~:text=It's%20because%20switchIfEmpty%20accepts%20Mono,mono's%20evaluation%20is%20already%20triggered.)

#### PR fixes following issue(s)
Fixes #25572 

#### Media
> A video or a GIF is preferred. when using Loom, don’t embed because it looks like it’s a GIF. instead, just link to the video
>
>
#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

## Testing
>
#### How Has This Been Tested?
> Please describe the tests that you ran to verify your changes. Also list any relevant details for your test configuration.
> Delete anything that is not relevant
- Not aware how to add an automated test case for this. 
>
>
#### Test Plan
> Add Testsmith test cases links that relate to this PR
>
>
#### Issues raised during DP testing
> Link issues raised during DP testing for better visiblity and tracking (copy link from comments dropped on this PR)
>
>
>
## How to test for QA
- Check if GSheet datasource authorization works on deploy preview - this can only work if the instance registration has succeeded cc: @prapullac 
## Checklist:
#### Dev activity


#### QA activity:
